pkg/api/v1/types.go

```diff
@@ -710,6 +725,22 @@ type Container struct {
 	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
 }
 
+func (ctr *Container) ApplyDefaults() {
```

Comment:
ApplyDefaults is maybe marginally better than what we have now, but I don't like the code in the interface definition file.

I'd like simple defaults and validation (e.g., min, max, regexp -- things that are in json schema) to be declarative, using struct tags or some other annotation mechanism.